.github/workflows/docker.yml

```diff
@@ -41,22 +49,34 @@ jobs:
           context: .
           file: ./docker/main.dockerfile
           build-args: DOCKER_META_VERSION=${{ steps.docker_meta_main.outputs.version }}
-          platforms: linux/amd64,linux/arm64,linux/386
+          platforms: ${{ fromJSON(env.DOCKER_PUSH) && 'linux/amd64,linux/arm64,linux/386' || 'linux/amd64' }}
           tags: ${{ steps.docker_meta_main.outputs.tags }}
           labels: ${{ steps.docker_meta_main.outputs.labels }}
-          push: true
+          push: ${{ fromJSON(env.DOCKER_PUSH) }}
+          load: ${{ !fromJSON(env.DOCKER_PUSH) }}
+      - name: Check images
+        run: |
+          df -h
+          docker image ls
+          docker image inspect ${{ steps.docker_meta_main.outputs.tags }}
+
       - name: Docker meta for cloud image
         id: docker_meta_cloud
         uses: crazy-max/ghaction-docker-meta@v1
         with:
           images: nbraun/dask-sql-cloud
+      - name: Check images_2
+        run: |
+          df -h
+          docker image ls
+          docker image inspect ${{ steps.docker_meta_main.outputs.tags }}
       - name: Build and push cloud image
         uses: docker/build-push-action@v2
         with:
           context: .
           file: ./docker/cloud.dockerfile
           build-args: DOCKER_META_VERSION=${{ steps.docker_meta_main.outputs.version }}
-          platforms: linux/amd64,linux/arm64,linux/386
+          platforms: ${{ fromJSON(env.DOCKER_PUSH) && 'linux/amd64,linux/arm64,linux/386' || 'linux/amd64' }}
           tags: ${{ steps.docker_meta_cloud.outputs.tags }}
           labels: ${{ steps.docker_meta_cloud.outputs.labels }}
-          push: true
+          push: ${{ fromJSON(env.DOCKER_PUSH) }}
```

Comment:
I'm not 100% sure but I think you might need to change this to be `load: true` in order for it to pick up the local base image. Then add an additional push step. Setting `load: true` changes the behaviour of the builder.

See https://stackoverflow.com/a/63927832/1003288